### ISSUE TITLE
Possible Memory Leak Issue

### ISSUE THREAD

**Issue by rcliao** [Automattic/kue#704]
# Introduction

Hi to the developers of kue module,

First of all, amazing npm module for handling message queue. It's simple and easy to get started; however I am currently facing a potential memory leak issue while creating a large amount of message jobs from a single method. In essence, my application server just crashes while creating such amount of jobs with the error message `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory`. Thus, I wonder if you as creator of Kue can help me out.

# Example Code

Here is an example test case of creating a large number of requests to crash:  
https://gist.github.com/rcliao/0d8c5a5173b9ceb1453b

In code, I am creating a large amount of message jobs while handling the message processing in the same application. 

# Result

Even though I'm testing with 1,000,000 jobs all at once to see what is the limit of server; nevertheless, with lower number of jobs (like 1000 jobs at once), server may not crash but with a peek of memory usage (up to 1.3GB for a while until the jobs are mostly being processed).

# Expectation

I expected the job variable in the for loop to be gone in memory by the time for loop finishes consider I never attach any job level events for the scalability concern.

# Some Finding for Option to Turn Off Job Events

I was looking around in this repository and found out that job will be kept in memory unless option `kue.createQueue( {jobEvents: false} )` is set with a related issue https://github.com/Automattic/kue/issues/401. But with this option, we still have a very high memory use when more jobs are created than the speed it is being processed. In other word, we created way more jobs than it is being processed; we may be facing a `process out of memory` fatal error to server crash.

# Summary

In summary, I wonder if there is any way to create job without the job sitting in memory while waiting to be processed in case we are creating more jobs than processing speed.

**Comment by rcliao** [comment]
As an update after a few more trials and errors, realized a simple mistake I was making in the for loop.

The biggest reason why there is memory spike is due to the fact that for loop with `job.save()` is asynchronous. Thus, I was creating all these jobs all at once.

With `eachLimit` method in `async` module, I am able to control over how many jobs being created at once before GC comes in and clean up; thus, resolving memory spike.

**Comment by misterjt** [comment]
#703 .Its senseless to write any issues.
